---
title: "EEMB_144L_2018_DADA2"
author: "Samantha Chen"
date: "11/18/2020"
output: github_document
---

```{r message=FALSE, warning=FALSE}
library(tidyverse)
library(dada2)
library(ShortRead)
```

```{r}
path <- "~/GitHub EEMB 144L/144l_students/Input_Data/week5/EEMB144L_2018_fastq/"

fnFs <- list.files(path, pattern = "_R1_001.fastq", full.names = TRUE)
fnRs <- list.files(path, pattern = "_R2_001.fastq", full.names = TRUE)
```

```{r}
FWD = "GTGYCAGCMGCCGCGGTAA"
REV = "GGACTACNVGGGTWTCTAAT"

all0rients <- function(primer) {
  require(Biostrings)
  dna <- DNAString(primer)
  orients <- c(Forward = dna, Complement = complement(dna), Reverse = reverse(dna), RevComp = reverseComplement(dna))
  return(sapply(orients, toString))
}

FWD.orients <- all0rients(FWD)
REV.orients <- all0rients(REV)
```

```{r}
primerHits <- function(primer, fn) {
  nhits <- vcountPattern(primer, sread(readFastq(fn)), fixed = FALSE)
  return(sum(nhits > 0))
}

rbind(FWD.ForwardReads = sapply(FWD.orients, primerHits, fn = fnFs[[1]]), 
      FWD.ReverseReads = sapply(FWD.orients, primerHits, fn = fnRs[[1]]), 
      REV.ForwardReads = sapply(REV.orients, primerHits, fn = fnFs[[1]]), 
      REV.ReverseReads = sapply(REV.orients, primerHits, fn = fnRs[[1]]))
```

## Forward Reads

```{r fig.height=10, fig.width=12}
plotQualityProfile(fnFs[1:12])
```

## Reverse Reads

```{r fig.height=12, fig.width=10}
plotQualityProfile(fnRs[1:12])
```
# Filtering and Trimming

```{r}
sample.names <- sapply(strsplit(basename(fnFs), "_L"), '[', 1)
sample.names

filt_path <- file.path(path, "filtered")
filtFs <- file.path(filt_path, paste0(sample.names, "_F_filt.fastq"))
filtRs <- file.path(filt_path, paste0(sample.names, "_R_filt.fastq"))
```

```{r}
out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen = c(250,150), maxN = 0, maxEE = c(2,2), truncQ = 2, rm.phix = TRUE, compress = TRUE)

out

```

# Error Rates and Plots

```{r}
errF <- learnErrors(filtFs, multithread = TRUE)
errR <- learnErrors(filtRs, multithread = TRUE)
```

```{r echo = FALSE, warning = FALSE, message = FALSE, fig.height = 10, fig.width = 12, fig.align = "center", warning = FALSE}
plotErrors(errF, nominalQ = TRUE)

```
In my forward reads, the estimated error rates are a good fit to the observed error rates. As expected, error rates drop as quality increases. Otherwise, the charts look okay and I can proceed.

```{r echo = FALSE, warning = FALSE, message = FALSE, fig.height = 10, fig.width = 12, fig.align = "center", warning = FALSE}
plotErrors(errR, nominalQ = TRUE)
```
In my reverse reads, the estimated error rates are also a good fit to the observed error rates. Similarly to the forward reads, error rates drop as quality increases. Otherwise, the charts look okay and I can proceed.

# Dereplication

```{r}
derepFs <- derepFastq(filtFs, verbose = TRUE)
derepRs <- derepFastq(filtRs, verbose = TRUE)

names(derepFs) <- sample.names
names(derepRs) <- sample.names
```

```{r}
dadaFs <- dada(derepFs, err = errF, multithread = TRUE)
dadaRs <- dada(derepRs, err = errR, multithread = TRUE)
```

